src/basic/stat-util.c

```diff
@@ -254,7 +254,8 @@ int fd_is_network_ns(int fd) {
         if (r <= 0)
                 return r;
 
-        if (ioctl(fd, NS_GET_NSTYPE) < 0)
+        r = ioctl(fd, NS_GET_NSTYPE);
```

Comment:
Thank you. This change has made the test pass. The only thing I don't understand is why the test passed for some time even without this commit. Could it be that `systemd-nspawn` failed to propagate the error somehow?

Reply:
The return value of `ioctl` must have been positive and `r` can't have been equal to `CLONE_NEWNET`, so `fd_is_network_ns` should have returned `0`, which should have made the test fail. I have no idea why it didn't fail when https://github.com/systemd/systemd/pull/8144 was opened.

Reply:
I think I figure it out. [`NS_GET_NSTYPE`](https://github.com/torvalds/linux/commit/e5ff5ce6e20ee22511398bb31fb912466cf82a36) was added relatively recently, so the return value of `ioctl` must actually have been negative because the machine where the test was running didn't support `NS_GET_NSTYPE`. The test started failing as soon as the kernel had been upgraded to `4.14`.

Reply:
Ah, interesting, I guess that makes a ton of sense then. I was as puzzled as you were...